### ISSUE TITLE
Can not create tasks on some user stories: 'NoneType' object has no attribute 'is_closed'

### ISSUE THREAD

**Issue by mlq** [taigaio/taiga-back#543]
Hello,

I can not create tasks on some user stories that have been promoted from an issue with the following exception:  

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/viewsets.py", line 81, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/views.py", line 420, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/taiga/taiga-back/taiga/base/api/views.py", line 417, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/mixins.py", line 72, in create
    self.object = serializer.save(force_insert=True)
  File "/usr/local/taiga/taiga-back/taiga/projects/notifications/mixins.py", line 243, in save
    obj = super(EditableWatchedResourceModelSerializer, self).save(**kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/serializers.py", line 1055, in save
    self.object = super().save(**kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/serializers.py", line 614, in save
    self.save_object(self.object, **kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/serializers.py", line 1073, in save_object
    obj.save(**kwargs)
  File "/usr/local/taiga/taiga-back/taiga/projects/tasks/models.py", line 86, in save
    return super().save(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/django/db/models/base.py", line 734, in save
    force_update=force_update, update_fields=update_fields)
  File "/usr/local/lib/python3.4/site-packages/django/db/models/base.py", line 758, in save_base
    update_fields=update_fields)
  File "/usr/local/lib/python3.4/site-packages/django/dispatch/dispatcher.py", line 189, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/local/taiga/taiga-back/taiga/projects/tasks/signals.py", line 102, in set_finished_date_when_edit_task
    if instance.status.is_closed and not instance.finished_date:
AttributeError: 'NoneType' object has no attribute 'is_closed'
```
I am using the latest commit from the stable branch.

Thank you!

**Comment by superalex** [comment]
Hello @mlq I can't reproduce the issue in tree.taiga.io (with the last stable release too), is it happening in all the promoted items or only in some of them?. Could you please check the status of the failing ones?

**Comment by mlq** [comment]
The status of a [failing one](https://bugs.pwmt.org/project/zathura/us/557?no-milestone=1) is set to 'New'. I can recall that at one point after the promotion its status was not set at all, but it is also happening to newly created user stories or user stories with different states (like in-progress) in the same project. So I think it has more to do with the project itself rather than the promoted user stories.

**Comment by superalex** [comment]
Could you please check the default status for that projects?, it's on the admin area -> project -> default values

Regards!,

**Comment by mlq** [comment]
Good call, not all of them were set which probably caused the missing state at one of the user stories. However, it still fails, but the backtrace is a bit different:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/viewsets.py", line 81, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/views.py", line 420, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/taiga/taiga-back/taiga/base/api/views.py", line 417, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/mixins.py", line 72, in create
    self.object = serializer.save(force_insert=True)
  File "/usr/local/taiga/taiga-back/taiga/projects/notifications/mixins.py", line 243, in save
    obj = super(EditableWatchedResourceModelSerializer, self).save(**kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/serializers.py", line 1055, in save
    self.object = super().save(**kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/serializers.py", line 614, in save
    self.save_object(self.object, **kwargs)
  File "/usr/local/taiga/taiga-back/taiga/base/api/serializers.py", line 1073, in save_object
    obj.save(**kwargs)
  File "/usr/local/taiga/taiga-back/taiga/projects/tasks/models.py", line 86, in save
    return super().save(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/django/db/models/base.py", line 734, in save
    force_update=force_update, update_fields=update_fields)
  File "/usr/local/lib/python3.4/site-packages/django/db/models/base.py", line 771, in save_base
    update_fields=update_fields, raw=raw, using=using)
  File "/usr/local/lib/python3.4/site-packages/django/dispatch/dispatcher.py", line 189, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/local/taiga/taiga-back/taiga/projects/references/models.py", line 101, in attach_sequence
    instance.save(update_fields=['ref'])
  File "/usr/local/taiga/taiga-back/taiga/projects/tasks/models.py", line 86, in save
    return super().save(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/django/db/models/base.py", line 734, in save
    force_update=force_update, update_fields=update_fields)
  File "/usr/local/lib/python3.4/site-packages/django/db/models/base.py", line 771, in save_base
    update_fields=update_fields, raw=raw, using=using)
  File "/usr/local/lib/python3.4/site-packages/django/dispatch/dispatcher.py", line 189, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/local/taiga/taiga-back/taiga/projects/tasks/signals.py", line 37, in try_to_close_or_open_us_and_milestone_when_create_or_edit_task
    _try_to_close_or_open_us_when_create_or_edit_task(instance)
  File "/usr/local/taiga/taiga-back/taiga/projects/tasks/signals.py", line 50, in _try_to_close_or_open_us_when_create_or_edit_task
    if us_service.calculate_userstory_is_closed(instance.user_story):
  File "/usr/local/taiga/taiga-back/taiga/projects/userstories/services.py", line 109, in calculate_userstory_is_closed
    if all([task.status.is_closed for task in user_story.tasks.all()]):
  File "/usr/local/taiga/taiga-back/taiga/projects/userstories/services.py", line 109, in <listcomp>
    if all([task.status.is_closed for task in user_story.tasks.all()]):
AttributeError: 'NoneType' object has no attribute 'is_closed
```

**Comment by superalex** [comment]
Now it seems related to the default status for "Tasks", could you check if it's set properly in the admin area?. I'm trying to have a project with invalid default values for status but I can't, do you remember doing something "special" to achieve it?

**Comment by mlq** [comment]
The default status for 'Tasks' is set to 'New' in the admin area.

Yes, I probably achieved this by importing a project without defining any default values

**Comment by superalex** [comment]
Could you update that default status no another one and then restore it to new again?.

Is it happening in all your imported projects or only in some of them?. Not having default values shouldn't be possible to achieve.

**Comment by mlq** [comment]
I tried that, but that did not change anything.

I've now created some example US in all other imported projects and it works in them without any problems. - At least it seemed to be possible to not pass any default values when I've migrated my projects (I've written a python tool that uses python-taiga to migrate from a roundup issue tracker to Taiga).

**Comment by superalex** [comment]
You are right @mlq 

We should review that piece of code, sorry for the inconvenience! ;)

**Comment by mlq** [comment]
No worries, do you know now what causes this? Can I resolve this somehow by updating the database, because at least there the is_closed fields seem to be set.

**Comment by superalex** [comment]
You could check at database level the tasks_task table to see if there are entries with null values for status_id

**Comment by mlq** [comment]
Thanks @superalex , there were several of those. Setting their status_id fixed the issue for now!

However, I've noticed that by fixing them other created tasks also showed up that were not in the list of broken ones. E.g.: I've had to patch 7 tasks, two of them called "ABC". After fixing them, around 6 "ABC" tasks showed up in that user story since I've tried to create that entry several times, but failed.
